### PR TITLE
Add the proto source file to the message adapter

### PR DESCRIPTION
**Pull request by kalfonso**
This PR adds a property to the Wire message adapter to point at the message's proto file path.

This change will be used to integrate Wire with Confluent's Schema Registry. Confluent's protobuf implementation uses the encoded message descriptors in the generated Java code. We need similar functionality in Wire. In order to do that  we need to access the proto file for a message. Wire already supports including the proto file in the classpath resources via ```protoLibrary = true``` configuration option. This change simplifies discovering the proto file for a message by adding it to the adapter.

Checks:
- [X] Code compiles correctly
- [X] Updated ```JavaGeneratorTest.generateAbstractAdapter``` which failed with this new change.
- [X] All tests passing

### CODE REVIEW

File: wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/ProtoAdapter.kt [hunk 1]
```diff
@@ -72,8 +73,9 @@ actual abstract class ProtoAdapter<E> actual constructor(
     type: Class<*>,
     typeUrl: String?,
     syntax: Syntax,
-    identity: E?
-  ) : this(fieldEncoding, type.kotlin, typeUrl, syntax, identity)
+    identity: E?,
+    sourceFile: String?,
+  ) : this(fieldEncoding, type.kotlin, typeUrl, syntax, identity, sourceFile)
```
[the remark this answers is not in the document]
Comment (kalfonso) [reply wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/ProtoAdapter.kt:78]: Thanks for the feedback. I'll make the changes and split the PR in 2

[the remark this answers is not in the document]
Comment (kalfonso) [reply wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/ProtoAdapter.kt:92]: Done

File: wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/ProtoAdapter.kt [hunk 2]
```diff
@@ -75,6 +76,11 @@ expect abstract class ProtoAdapter<E>(
    */
   val identity: E?
 
+  /**
+   * Path to the proto file.
```
[the remark this answers is not in the document]
Comment (kalfonso) [reply wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/ProtoAdapter.kt]: Done

File: wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 3]
```diff
@@ -1329,6 +1330,7 @@ class KotlinGenerator private constructor(
         .addSuperclassConstructorParameter("\n%M",
             MemberName(Syntax::class.asClassName(), type.syntax.name))
         .addSuperclassConstructorParameter("\nnull\n⇤")
+        .addSuperclassConstructorParameter("\n%S", type.location.path)
```
[the remark this answers is not in the document]
Comment (kalfonso) [reply wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: Done

**Comment by kalfonso** [comment]
> That'd help if there were 2 commits, one for the change, and one for the re-generated golden files.

@oldergod Split PR in two commits for better clarity.

File: wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 4]
```diff
@@ -1328,7 +1329,8 @@ class KotlinGenerator private constructor(
         .addSuperclassConstructorParameter("\n%S", type.type.typeUrl!!)
         .addSuperclassConstructorParameter("\n%M",
             MemberName(Syntax::class.asClassName(), type.syntax.name))
-        .addSuperclassConstructorParameter("\nnull\n⇤")
+        .addSuperclassConstructorParameter("\nnull⇤")
```
Comment (swankjesse) [inline wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt:1332]: nit: can you move the `⇤` to the next line, after the `\n` character? That’ll fix the issue where it’s unindented one line too early.

**Review by swankjesse — APPROVED** [review]

**Comment by swankjesse** [comment]
I pushed a fix for my nitpick & regenerated tests.

File: wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt (hunk 4 above)
Comment (swankjesse) [reply wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: Fixed.

**Pull request merged**